search/roadmap.adoc

```diff
@@ -26,34 +26,46 @@ Analyzers::
 A new DSL will be introduced to allow people to define Analyzers programmatically at boostrap, as an alternative to the current `@Analyzer` and `@AnalyzerDef` annotations.
 We might want to automatically register a set of commonly used Analyzers - let us know what you think of this as the choice is hard!
 
+Java 9 compatibibility::
+We're aiming at regularly testing compatibility with Java 9 on top of the regular tests on Java 8.
+Testing will be limited to having Hibernate Search on the regular classpath: running as a module is an additional step we deferred to version 5.9.
+
+
+== Hibernate Search 5.9
+
+Elasticsearch performance::
+We applied some interesting performance improvements in version 5.8 but there are some more interesting ideas which could provide
+a radical boost. Moved these to 5.9 as they are quite complex and need fleshing out.
```

Comment:
Are you talking about taking advantage of [optimistic concurrency control in Elasticsearch](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-index_.html#index-versioning) using the "version" parameter? If so, we should probably explain that: as it is, the comment is not very informative, and we know that we won't try to find other ways of improving performance because 5.9 should ship as fast as possible (we don't want to waste our time).
Also, in my mind the change would be more about preventing version conflicts than about improving performance, but performance improvements would of course be welcome.

Reply:
Yes that's what I'm referring to but we didn't flesh it all at all - we might need to do something entirely different. A roadmap is not meant to be detailed, especially not on implementation details, more like a general hing that if you don't like performance today don't worry we're still hoping to work on it.

On the speed: it's a good point but we can always split things up into multiple releases or simply can this plan replacing it with some simpler optimisations *as long as we don't make specific promises*